### PR TITLE
Feature/libtcg

### PR DESCRIPTION
**Pull request by AntonJohansson**

### CODE REVIEW

File: include/libtcg/libtcg.h [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define LIBTCG_INSN_MAX_ARGS 16
+#define LIBTCG_MAX_NAME_LEN 32
+
+/*
+ * We start out with a buch of constants and enums taken from
```
Comment (zTehRyaN) [inline include/libtcg/libtcg.h]: Just a typo.

`s/buch/bunch`

File: libtcg/dump_tinycode_instruction.c [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+                                   insn->nb_oargs);
+        for (uint32_t i = 0; i < insn->nb_oargs; i++) {
+            fmt_append_to_stringbuffer(&buffer, ",%s",
+                                       insn->output_args[i].temp->name);
+        }
+        for (uint32_t i = 0; i < insn->nb_iargs; i++) {
+            fmt_append_to_stringbuffer(&buffer, ",%s",
+                                       insn->input_args[i].temp->name);
+        }
+        //        for (i = 0; i < nb_iargs; i++) {
```
Comment (zTehRyaN) [inline libtcg/dump_tinycode_instruction.c]: What was this for? Residual debug code?

File: libtcg/libtcg.c [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+         * Here we handle `temp` arguments so output and input args.
+         * Note: `insn.args[i]` and `op->args[i]` may have different
+         * integer sizes.
+         */
+        for (uint32_t i = 0; i < insn.nb_oargs; ++i) {
+            TCGTemp *ts = arg_temp(op->args[i]);
+            int idx = temp_idx(ts);
+            /*
+             * TODO(anjo): Here we are casting between TCG's enums and ours.
+             * This can ofcourse cause problems. I am here assuming that the
```
Comment (zTehRyaN) [inline libtcg/libtcg.c]: Just a typo.

`s/ofcourse/of course`

File: libtcg/libtcg.c [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+                .temp = temp,
+            };
+        }
+
+        for (uint32_t i = 0; i < insn.nb_iargs; ++i) {
+            TCGTemp *ts = arg_temp(op->args[insn.nb_oargs + i]);
+            int idx = temp_idx(ts);
+            /*
+             * TODO(anjo): Here we are casting between TCG's enums and ours.
+             * This can ofcourse cause problems. I am here assuming that the
```
Comment (zTehRyaN) [inline libtcg/libtcg.c]: Just a typo.

`s/ofcourse/of course`

File: libtcg/dump_tinycode_instruction.c (hunk 2 above)
Comment (AntonJohansson) [reply libtcg/dump_tinycode_instruction.c]: The comment is a piece of code from the original function in `tcg.c` which I kept as a reference when translating it to using the `fmt_append_to_stringbuffer` function and the `libtcg.h` data. It's been removed! :)

**Pull request closed without merging**